geth/jail/cell.go

```diff
@@ -68,7 +89,9 @@ func (c *Cell) Stop() {
 // async call, like callback.
 func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
 	task := looptask.NewCallTask(fn, args...)
-	c.lo.Add(task)
-	// TODO(divan): review API of `loop` package, it's contrintuitive
-	go c.lo.Ready(task)
+	// Add a task to the queue.
+	c.loop.Add(task)
+	// And run the task immediatelly.
+	// It's a blocking operation.
+	c.loop.Ready(task)
```

Comment:
What happens if loop's queue is filled and this call will block until queue (buffered channel) has some free space for new items? Wouldn't it be better to trade this for goroutines?

Reply:
I think it's not even buffered now. Using a goroutine can introduce some hazards and it will make it hard to debug if one task will block others from being processed, for instance.

Also, `CallAsync` now is called only from a goroutine so nothing changes if I remove `go` from this line.

I should change the loop API a bit then:
1. Close `l.ready` channel when the loop is stopped,
2. `func (l *Loop) Ready(t Task)` should be `func (l *Loop) Ready(t Task) error` + should cancel a task in case of an error (closed loop will yield an error),
3. Maybe `CallAsync` should take a context? And fail when it's canceled? To prevent handing forever in some strange case.

Reply:
Added a separate issue https://github.com/status-im/status-go/issues/415